core/trino-main/src/main/java/io/trino/operator/WorkProcessorSourceOperatorAdapter.java

```diff
@@ -176,9 +176,9 @@ public boolean isFinished()
     public void close()
             throws Exception
     {
-        operatorContext.setLatestMetrics(sourceOperator.getMetrics());
-        operatorContext.setLatestConnectorMetrics(sourceOperator.getConnectorMetrics());
         sourceOperator.close();
+        operatorContext.setLatestMetrics(sourceOperator.getMetrics());
```

Comment:
there is also `WorkProcessorPipelineSourceOperator` where we don't set metrics after source is closed. Also some test would be nice.

Reply:
Thanks, added in https://github.com/trinodb/trino/pull/9615/commits/a6daa33744acbd03a52c82d6e95a04c69c4cc0f8.